### DISCUSSION TITLE
Do the parameters not support Like queries?

### DISCUSSION THREAD

**Discussion by ghost** [rusqlite/rusqlite#1108] [Questions] [answered]
Thank you very much for creating this library, it works very well, but I recently encountered a problem with fuzzy queries.

This is my code, sql statement tested is available. 
```rust
    pub fn find_word_by_regexp_keyword(
        &self,
        keyword_reg: &str,
    ) -> Result<Vec<Word>, rusqlite::Error> {
        let mut stmt = self
            .connection
            .prepare(r#"SELECT keyword FROM word WHERE keyword LIKE ":keyword_reg%";"#)?;
        let words = stmt
            .query_map::<Word, _, _>(&[(":keyword_reg", &keyword_reg)], |row| {
                self.find_word_by_keyword(&row.get::<_, String>(0).unwrap())
            })?
            .collect::<Result<Vec<Word>, _>>()?;
        Ok(words)
    }
```
With the above code, I get an error:
```shell
Err: InvalidParameterName(
    ":keyword_reg",
)
```
My sqlite version is : 3.37.0 2021-11-27 14:13:22

**Comment by gwenn** [comment, marked as answer]
This is not related to `rusqlite`.
You should be able to reproduce your issue with SQLite shell (see `.parameter` [here](https://sqlite.org/cli.html)).
And not related but you should use single quote for string literal.
For me, you need to do:
```sql
SELECT keyword FROM word WHERE keyword LIKE (:keyword_reg || '%');
```

**Comment by gwenn** [comment]
See #573 for a similar issue.
